app/models/user.rb

```diff
@@ -1,2 +1,9 @@
 class User < ApplicationRecord
+  has_secure_password
+  has_many :articles
+  validates :email, presence: true, uniqueness: true
+  validates :email, format: { with: URI::MailTo::EMAIL_REGEXP }
```

Comment:
Where is this constant `EMAIL_REGEXP` defined?

Reply:
EMAIL_REGEXP is constant built into URI in a standard ruby library